packages/integrations/vercel/src/serverless/entrypoint.ts

```diff
@@ -10,6 +10,10 @@ export const createExports = (manifest: SSRManifest) => {
 	const handler = async (req: IncomingMessage, res: ServerResponse) => {
 		const clientAddress = req.headers['x-forwarded-for'] as string | undefined;
 		const localsHeader = req.headers[ASTRO_LOCALS_HEADER];
+		const realPath = req.headers['x-astro-path'];
+		if (typeof realPath === 'string') {
+			req.url = realPath;
+		}
```

Comment:
This is so that the edge function can call the serverless function in `next()`.

Reply:
I still think it's worth adding a comment because the read and write of the header happens in two different places, so it could be helpful to give some context.

Reply:
Added

https://github.com/withastro/astro/pull/9585/files#diff-071ccfbdfd7b9bc803219179f88ae0bdb227a399bc4a481b7831c82b801d7a17R32-R43